### PR TITLE
Morgan's D04 Homework

### PR DESCRIPTION
**Pull request by sheffieldjordan**
Here's my homework from Day 04!

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -11,18 +11,36 @@
 #         - then ends the program
 ################################################################################
 # Imports
-
+import random
 
 # Body
-
+number = random.randrange(1,25)
```
Comment (danielsgriffin) [inline D04_ex00.py]: Is this right? 1 through 25 inclusive ?

File: HW04_ch07_ex02.py [hunk 2]
```diff
@@ -21,15 +21,28 @@
...
+			while problem != 'done':
+				problem = input('What do you want to evaluate?>')
+				result = eval(problem)
+				if problem == 'done':
+					return
+				print(result)
+			print("You're done!")
+			print("Your last result was", result)
+			break
+		except:
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: Ask me if you cannot figure it out w/ your partner this morning.

File: HW04_ch08_ex04.py [hunk 3]
```diff
@@ -14,17 +14,20 @@
 
 
 def any_lowercase1(s):
-    """Explain what is wrong, if anything, here.
+    """This function returns boolean values, meaning it will stop running once either
```
Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: Correct, but what is the goal?

File: HW04_ch08_ex05.py [hunk 4]
```diff
@@ -2,3 +2,27 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]:
```
Book Examples:
  X  'A'     by 3   => got: 'd'     expected: 'D'
  X  'Z'     by 1   => got: 'a'     expected: 'A'
 OK  'cheer' by 7   => got: 'jolly' expected: 'jolly'
 OK  'melon' by -10 => got: 'cubed' expected: 'cubed'
  X  'IBM'   by -1  => got: 'hal'   expected: 'HAL'

Daniel's Tests:
 OK  'abc'   by 1   => got: 'bcd'   expected: 'bcd'
 OK  'xyz'   by 1   => got: 'yza'   expected: 'yza'
  X  'ABC'   by 1   => got: 'bcd'   expected: 'BCD'
  X  'XYZ'   by 1   => got: 'yza'   expected: 'YZA'
  X  'Az'    by 1   => got: 'ba'    expected: 'Ba'
  X  'Aa'    by 26  => got: 'aa'    expected: 'Aa'
 OK  'abc'   by -1  => got: 'zab'   expected: 'zab'
 OK  'xyz'   by -1  => got: 'wxy'   expected: 'wxy'
  X  'ABC'   by -1  => got: 'zab'   expected: 'ZAB'
  X  'XYZ'   by -1  => got: 'wxy'   expected: 'WXY'
  X  'Az'    by -1  => got: 'zy'    expected: 'Zy'
  X  'Aa'    by -26 => got: 'aa'    expected: 'Aa'
```

File: HW04_ch08_ex05.py [hunk 5]
```diff
@@ -2,3 +2,27 @@
...
+		x = (ord(letter)) + rotated_value
+		if x < ord('a'):
+			x = x + 26
+		elif x > ord('z'):
+			x = x - 26
+		rotated_word += (chr(x))
+	
+	return rotated_word
+
+def main():
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: Good test examples, but did you check against expected/desired values?

**Comment by danielsgriffin** [comment]
Good job. Pay attention to those tests.

File: HW04_string2.py [hunk 6]
```diff
@@ -18,7 +18,13 @@
 # Return the resulting string.
 def verbing(s):
     # +++your code here+++
-    return
+    if len(s) >= 3:
+        if s[-3:] == 'ing':
+            s = s + "ly"
+        else: 
+            s += 'ing'
```
Comment (michellecarney) [inline HW04_string2.py]: Nice :)

File: HW04_string2.py [hunk 7]
```diff
@@ -18,7 +18,13 @@
...
+    return s
+
```
Comment (shannonh90) [inline HW04_string2.py]: good job

File: HW04_string2.py [hunk 8]
```diff
@@ -31,7 +37,11 @@ def verbing(s):
 # This dinner is good!
 def not_bad(s):
     # +++your code here+++
-    return
```
Comment (shannonh90) [inline HW04_string2.py]: do you need this return? honest question!

File: HW04_string2.py [hunk 9]
```diff
@@ -43,7 +53,20 @@ def not_bad(s):
...
+    else:
+        a_half = a_len//2 + 1
+    if b_len % 2 == 0:
+        b_half = b_len/2
+    else:
+        b_half = b_len//2 + 1
+
+    front_back = a[:a_half] + b[:b_half] + a[a_half:] + b[b_half:] 
+
+    return front_back
```
Comment (shannonh90) [inline HW04_string2.py]: looks good!

File: HW04_string2.py [hunk 10]
```diff
@@ -43,7 +53,20 @@ def not_bad(s):
 #  a-front + b-front + a-back + b-back
 def front_back(a, b):
     # +++your code here+++
-    return
+    a_len = len(a)
+    b_len = len(b)
+    if a_len % 2 == 0:
+        a_half = a_len/2
+    else:
+        a_half = a_len//2 + 1
```
Comment (michellecarney) [inline HW04_string2.py]: nice use of floor division :)

File: HW04_string2.py [hunk 11]
```diff
@@ -43,7 +53,20 @@ def not_bad(s):
...
+    if b_len % 2 == 0:
```
Comment (michellecarney) [inline HW04_string2.py]: Code won't run b/c two if statements, but nest if statements into each other
if a mod 2 == 0
>>if b mod 2 ==0
>>else #for b mod 2 != 0
else #for a mod 2 != 0 
>>(copy b from above)

**Pull request closed without merging**